src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -143,7 +172,7 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 					domain = "connect.monstercat.com"
 					path = "/"
 					BasicCookieStore().also { it.addCookie(this) }
-				})
+				}.apply { cookieStore = this })
```

Comment:
I would like to keep this method pure, i.e. no side effects.

Reply:
How could I get the cookieStore then ?

Reply:
I'm sure you can find a solution ;) otherwise you have to split it up or something. But a method starting with `create` should not modify anything, solely create.

Reply:
Using HttpClientContext here : 6cc64ba6d2ca7d3e42403dbefb28b93df5cd7f29

Reply:
have you tested this? But it definitely looks better to remove this class-wide variable!

Reply:
Yes, works great.